qiskit/transpiler/passes/scheduling/base_scheduler.py

```diff
@@ -261,10 +261,21 @@ def _get_node_duration(
     ) -> int:
         """A helper method to get duration from node or calibration."""
         indices = [dag.find_bit(qarg).index for qarg in node.qargs]
+        is_mid_circuit = any(
```

Comment:
I would suggest moving this inside the `elif isinstance(node.op, Measure)` on L275 as we only need to do this if we're evaluating a measurement. This method will get called a lot more frequently for non-measurement operations so this is just extra overhead in those cases.